source/js/foundation/template-js-handler/window/sticky-share-button-group.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Bind scroll handler to window to toggle sticky share buttons on blog page
+ */
+export default () => {
+  let blogPageStickyButtons = document.querySelector(
+    `#view-blog .blog-sticky-side .share-button-group`
+  );
+  let blogPageFullButtons = document.querySelector(
+    `#view-blog .blog-body .share-button-group`
+  );
+
+  if (blogPageStickyButtons && blogPageFullButtons) {
+    const isInViewport = element => {
+      let box = element.getBoundingClientRect();
+
+      return box.top <= window.innerHeight && box.top + box.height >= 0;
+    };
+
+    const toggleStickyButtons = () => {
+      if (isInViewport(blogPageFullButtons)) {
+        blogPageStickyButtons.classList.add(`faded`);
+      } else {
+        blogPageStickyButtons.classList.remove(`faded`);
+      }
+    };
+
+    window.addEventListener(`scroll`, toggleStickyButtons);
```

Comment:
scroll events need a  `{ passive: true }` as third arg, to prevent jankiness on mobile.

(it's a relatively new addition to the event system specifically to make scrolling better for low-resource devices like old computers are mobile devices)

Reply:
Nice!